internal/controller/datadogagent/controller_reconcile_dca.go

```diff
@@ -30,66 +30,103 @@ import (
 	"github.com/DataDog/datadog-operator/internal/controller/datadogagent/override"
 	"github.com/DataDog/datadog-operator/pkg/condition"
 	"github.com/DataDog/datadog-operator/pkg/constants"
+	"github.com/DataDog/datadog-operator/pkg/controller/utils"
 	"github.com/DataDog/datadog-operator/pkg/controller/utils/datadog"
 	"github.com/DataDog/datadog-operator/pkg/kubernetes"
 )
 
-func (r *Reconciler) reconcileV2ClusterAgent(logger logr.Logger, requiredComponents feature.RequiredComponents, features []feature.Feature, dda *datadoghqv2alpha1.DatadogAgent, resourcesManager feature.ResourceManagers, newStatus *datadoghqv2alpha1.DatadogAgentStatus) (reconcile.Result, error) {
+func (r *Reconciler) reconcileV2ClusterAgent(ctx context.Context, logger logr.Logger, requiredComponents feature.RequiredComponents, features []feature.Feature, dda *datadoghqv2alpha1.DatadogAgent, resourcesManager feature.ResourceManagers, newStatus *datadoghqv2alpha1.DatadogAgentStatus) (reconcile.Result, error) {
 	var result reconcile.Result
 	now := metav1.NewTime(time.Now())
 
-	// Start by creating the Default Cluster-Agent deployment
-	deployment := componentdca.NewDefaultClusterAgentDeployment(dda.GetObjectMeta(), &dda.Spec)
-	podManagers := feature.NewPodTemplateManagers(&deployment.Spec.Template)
-
-	// Set Global setting on the default deployment
-	global.ApplyGlobalSettingsClusterAgent(logger, podManagers, dda.GetObjectMeta(), &dda.Spec, resourcesManager, requiredComponents)
-
-	// Apply features changes on the Deployment.Spec.Template
-	var featErrors []error
-	for _, feat := range features {
-		if errFeat := feat.ManageClusterAgent(podManagers); errFeat != nil {
-			featErrors = append(featErrors, errFeat)
+	// Get provider list for introspection
+	providerList := map[string]struct{}{kubernetes.LegacyProvider: {}}
+	if r.options.IntrospectionEnabled {
+		nodeList, err := r.getNodeList(ctx)
+		if err != nil {
+			return reconcile.Result{}, err
 		}
+		providerList = kubernetes.GetProviderListFromNodeList(nodeList, logger)
+		logger.Info("providerList for cluster agent", "providerList", providerList) //output is default-rhcos for label node.openshift.io/os_id=rhcos
 	}
-	if len(featErrors) > 0 {
-		err := utilerrors.NewAggregate(featErrors)
-		updateStatusV2WithClusterAgent(deployment, newStatus, now, metav1.ConditionFalse, "ClusterAgent feature error", err.Error())
-		return result, err
-	}
+	// TODO: does it make sense to put provider list BEFORE reconciler? this is where the real reconcilation starts
+	// Reconcile cluster agent for each provider
+	var errs []error
+	for provider := range providerList {
```

Comment:
Would this create multiple DCA deployments if we have two providers (like cos and ubuntu)? 

Can we assume that only one provider will affect it's config, given DCA is responsible for cluster-level resources/telemetry and technically we can't have two DCA deployments.